src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -308,6 +308,26 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                 packageCount: _packageCount,
                 noOpProjectsCount: _noOpProjectsCount,
                 upToDateProjectsCount: _upToDateProjectCount,
+                unknownProjectsCount: sortedProjects
+                    .Where(x => x.ProjectStyle == ProjectStyle.Unknown).Count(),
```

Comment:
I know this reads better but the number of enumerations here is scary. Can we just do it in 1? :)

Reply:
I will change them to single line assignments and use a dictionary to avoid multiple iterations on the list.